core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/SatisfiesExpression.kt

```diff
@@ -0,0 +1,83 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.relational
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.SatisfiesType.ANY
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.SatisfiesType.EVERY
+import ch.ergon.dope.resolvable.expression.unaliased.type.toArrayType
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.BooleanType
+import ch.ergon.dope.validtype.ValidType
+
+object IteratorManager {
+    var count: Int = 1
+        get() = field++
+
+    fun resetCounter() {
+        count = 1
+    }
+}
+
+const val DEFAULT_ITERATOR_VARIABLE = "iterator"
+
+enum class SatisfiesType {
+    ANY,
+    EVERY,
+}
+
+sealed class SatisfiesExpression<T : ValidType>(
+    private val satisfiesType: SatisfiesType,
+    private val list: TypeExpression<ArrayType<T>>,
+    private val variable: String,
+    private val predicate: (Iterator<T>) -> TypeExpression<BooleanType>,
+) : TypeExpression<BooleanType> {
+    override fun toDopeQuery(): DopeQuery {
+        val listDopeQuery = list.toDopeQuery()
+        val iteratorVariable = if (variable == DEFAULT_ITERATOR_VARIABLE) variable + IteratorManager.count else variable
```

Comment:
This check looks a bit hacky, can't you already generate the correct name when entering the consturctor:
```
class AnySatisfiesExpression<T : ValidType>(
    list: TypeExpression<ArrayType<T>>,
    variable: String = DEFAULT_ITERATOR_VARIABLE + IteratorManager.count,
    predicate: (Iterator<T>) -> TypeExpression<BooleanType>,
) : SatisfiesExpression<T>(ANY, list, variable, predicate)
```
Or even better, move the generation into the iteratorManager:
```
    val defaultIteratorName: String
        get() = "iterator$count"
```

Reply:
I added it to the manager.